### ISSUE TITLE
"Add to playlist" not functional w/ Navidrome

### ISSUE THREAD

**Issue by whorfin** [jeffvli/sonixd#371]
**Describe the bug**
Using the latest (0.15.3) on windows, connected to a latest Navidrome server, "Add to playlist" does not work.

**To Reproduce**
1. Select a number of tracks in eg "Now Playing"
2. Right-mouse and "Add to playlist"
3. Select an existing playlist 
4. Click "Add"
5. Note the toast which says "Added ___  item(s) to playlist ___"
6. Select the playlist used above
7. Note the playlist is empty/does not contain new tracks

**Expected behavior**
The playlist should contain the newly added tracks, but it does not.
Deleting and creating playlists works.
Attempting to add tracks does not
Turning up Navidrome log level, I can see that the toast for success appears before any action on the server, and the server logs no attempt to interact with playlists, only a "/rest/getPlaylists.view" call
Adding to playlists with the Navidrome web client works fine, and i see the expected POST to "api/playlist/____/tracks"

**Desktop (please complete the following information):**
 - OS: Windows 10 x64
 - Version 0.15.3

**Comment by jeffvli** [comment]
I'm unable to reproduce this.

Can you use either [Debugtron](https://github.com/bytedance/debugtron) or run the app locally to check the browser console for errors?

**Comment by whorfin** [comment]
hmmm, ok
i created a new playlist called "test" and the first time, things seem to work
then i deleted its contents, which seemed to work
until i navigated away and came back, then it still had previous contents
at that point, trying to add more tracks [like 24] would do the same silent fail - seemed to work until i navigate to the playlist
Looking in the Debugtron debugger window, there were no messages other than the initial info on sonixd start on the version update check

**Comment by jeffvli** [comment]
If you make any changes in the playlist page, you need to click the "Save" button in the header after making the changes. If you're using Navidrome, there's an issue with deleting all the tracks in a playlist at once. There needs to be at least 1 track in the playlist when saving, otherwise the save will fail.

https://user-images.githubusercontent.com/42182408/182704646-0e5191aa-0c05-4c0e-96de-e6bb8762e791.mp4

I added a video of me trying to reproduce your steps.
1. Create playlist
2. Go to "Now Playing" and add songs to the new playlist
3. Delete all songs from playlist (and save)
4. Leave and re-navigate to the playlist page (All songs are still here due to Navidrome bug. Needs at least 1 track to save)
5. Delete all songs from the playlist except 1
6. Leave and re-navigate to the playlist page (Changes are now applied, only 1 track left)
7. Go to "Now Playing" and add new songs to the playlist
8. New songs are now visible

Though this doesn't explain why no network requests are being sent for the playlist adds.
Did you install Sonixd via the release binary? If not, uninstall and re-install it using that instead of any alternative methods.

You can also try to reset your settings file completely:

1. Close Sonixd
2. Navigate to `%appdata%/sonixd` and rename the `settings.json` file to anything like `settings.json.backup`
3. Re-open Sonixd and confirm that the `settings.json` file was recreated.

**Comment by whorfin** [comment]
Thank you for that "must have at least one entry" Navidrome tip!
I did not know that; that helps with deleting from existing playlists.  FWIW deleting all entries works just fine in the Navidrome web-app.

I am running the binary from the "releases".

Here is what happens on the "adding" fail for me:

https://user-images.githubusercontent.com/1237590/182714661-eb6b2853-de50-4c3a-95f9-e7e98f9e3570.mp4

Here's an interesting observation though:
If I attempt to add 14 tracks [as in your example], it works fine.
In the video I am trying to add 25 tracks, and this fails.
I tried some more sizes in a hunt...
24 also fails, 20 fails, 19 fails, 18 and 17 worked

**Comment by jeffvli** [comment]
I'm unable to reproduce that as well. I'm able to add tracks regardless of how many.
I also just noticed that right click isn't working when debugging the app using Debugtron, so I just built a production version that has the console open.

If you go to the [release page](https://github.com/jeffvli/sonixd/releases/tag/v0.15.3), download the `Sonixd-0.15.3-win-x64-DEBUG.zip` file and install that version (it won't override your settings). You should be able to more easily debug what's going on then.

**Comment by whorfin** [comment]
Hoorah!
🍾 
Thank you so much for the debug build, I was able to figure it out:
The length was a clue; the console clearly showed a 414 error on the `/rest/getPlaylists.view` call.  I had not seen this previously when running under Debugtron.
But this explains why Navidrome never saw the call - the 414 was happening because my reverse proxy for SSL had been bolted down a little too tightly.  I relaxed the max URI limits and everything is happy.
FYI: while everything worked out nicely, the debug build throws this error on launch:
![image](https://user-images.githubusercontent.com/1237590/182747714-265100fd-3ccf-4cbf-9cb8-84fe9129735e.png)
